R/RunAnalyses.R

```diff
@@ -493,24 +500,30 @@ runCmAnalyses <- function(connectionDetails,
   }
 
   # Trimming/Matching/Stratifying -----------------------------------------
-  subset <- referenceTable[!duplicated(referenceTable$strataFile), ]
-  subset <- subset[subset$strataFile != "", ]
-  subset <- subset[!file.exists(file.path(outputFolder, subset$strataFile)), ]
+  subset <- referenceTable[!duplicated(referenceTable$strataFile),]
+  subset <- subset[subset$strataFile != "",]
+  subset <- subset[!file.exists(file.path(outputFolder, subset$strataFile)),]
   if (nrow(subset) != 0) {
     message("*** Trimming/Matching/Stratifying ***")
+
     createTrimMatchStratTask <- function(i) {
-      refRow <- subset[i, ]
+      refRow <- subset[i,]
       analysisRow <- ParallelLogger::matchInList(
         cmAnalysisList,
         list(analysisId = refRow$analysisId)
       )[[1]]
-      task <- list(
-        psFile = file.path(outputFolder, refRow$psFile),
-        args = analysisRow,
-        strataFile = file.path(outputFolder, refRow$strataFile)
-      )
-      return(task)
+
+      if (file.exists(file.path(outputFolder, refRow$psFile))) {
```

Comment:
Why the condition? Is there something wrong with the code for generating the reference table that will specify strataFiles to be created without there being psFiles? Maybe best to fix it there?